code/game/machinery/cryopod.dm

```diff
@@ -268,37 +268,19 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 			for(var/datum/mind/M in O.team.members)
 				to_chat(M.current, "<BR><span class='userdanger'>Your target is no longer within reach. Objective removed!</span>")
 				M.announce_objectives()
-		else if(O.target && istype(O.target, /datum/mind))
-			if(O.target == mob_occupant.mind)
-				var/old_target = O.target
-				O.target = null
-				if(!O)
-					return
-				O.find_target()
-				if(!O.target && O.owner)
-					to_chat(O.owner.current, "<BR><span class='userdanger'>Your target is no longer within reach. Objective removed!</span>")
-					for(var/datum/antagonist/A in O.owner.antag_datums)
-						A.objectives -= O
-				if (!O.team)
-					O.update_explanation_text()
-					O.owner.announce_objectives()
-					to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
-				else
-					var/list/objectivestoupdate
-					for(var/datum/mind/own in O.get_owners())
-						to_chat(own.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
-						for(var/datum/objective/ob in own.get_all_objectives())
-							LAZYADD(objectivestoupdate, ob)
-					objectivestoupdate += O.team.objectives
-					for(var/datum/objective/ob in objectivestoupdate)
-						if(ob.target != old_target || !istype(ob,O.type))
-							return
-						ob.target = O.target
-						ob.update_explanation_text()
-						to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
-						ob.owner.announce_objectives()
+		else if(O.target == mob_occupant.mind)
+			O.target = null
+			O.find_target()
+			O.update_explanation_text()
+
+			var/list/owners = O.get_owners()
+
+			if(!O.target && O.owner)
 				qdel(O)
```

Comment:
This simple qdel works *because* Yogs added `/datum/objective/Destroy` at the same PR:

https://github.com/JamieD1/Yogstation/blob/d1e9753344b2e11297e79b3c179d08c68a55cdb3/code/game/gamemodes/objective.dm#L20-L27

Therefore, you should *also* port that.